Add registrationTokens alias (as per new GCM terminology)

eladnava opened the pull request:
Create an alias for `registrationTokens` as per the new GCM terminology "registration tokens" instead of "registration IDs". 

```
// Now the sender can be used to send messages
sender.send(message, { registrationTokens: regTokens }, function (err, result) {
    if(err) console.error(err);
    else    console.log(result);
});
```
It behaves the same as `registrationIds`. We decided to leave `registrationIds` intact to maintain backward compatibility.

Also, update the "no recipient" error message, as recipients can be both tokens and topics now.

```
 if (Array.isArray(recipient) && !recipient.length) {
        debug('No recipient provided!');
        return process.nextTick(callback.bind(this, 'No recipient provided!', null));
    }
```
Finally, incorporate the new alias in the senderSpec.js test.

References #160.

hypesystem commented:
There should be a test for the `registrationIds` key, as well, for backwards compatibility :-) Other than that, it looks good, and I am ready to merge :)

eladnava commented:
@hypesystem how would you like to test this?

The current test utilizes `registrationTokens` and checks whether the registration_ids are set correctly (we need to rename that field as well).

Would you like to duplicate this test and replace the `registrationTokens` parameter with `registrationIds`? Or did you have something else in mind?

hypesystem commented:
I would simply duplicate the test, as you suggest :-)

The `registration_ids` field is the one used in GCM, so we can't really change the name of that: https://developers.google.com/cloud-messaging/http-server-ref#table1

eladnava commented:
@hypesystem Excellent, added an additional test. :+1: 

By the way, I found some other references in the package to "registration IDs", or use of a variable called "regIds" -- would you like me to refactor those? (Leaving the 'registration_ids' GCM parameter intact, of course)

hypesystem commented:
Yeah, sure, let's get them all for consistency :)

eladnava commented:
Excellent, renamed all occurrences I could find and split up the modifications into 4 commits (1 per file). Let me know what you think @hypesystem :+1:

hypesystem commented:
Looks good! Thanks for the great work :-) :+1:
